### PR TITLE
fix: strip region from redis cache keys

### PR DESCRIPTION
**Pull request by jbee**
Forgot to strip the region from the cache keys when these are exposed as `keys()`.

### CODE REVIEW

**Review by ameenhere — APPROVED** [review]

**Review by larshelge — APPROVED** [review]

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/RedisCache.java [hunk 1]
```diff
@@ -148,8 +148,10 @@ public V get( String key, Function<String, V> mappingFunction )
     @Override
     public Set<String> keys()
     {
-        var keys = redisTemplate.keys( cacheRegion + "*" );
-        return keys == null ? emptySet() : unmodifiableSet( keys );
+        var keys = redisTemplate.keys( cacheRegion + ":*" );
+        return keys == null
+            ? emptySet()
+            : keys.stream().map( key -> key.substring( key.indexOf( ':' ) + 1 ) ).collect( toSet() );
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/RedisCache.java:154]: Is there always a ":" in the keys, and what if there is not? Should we handle that case just to be defensive? And perhaps move this logic to a separate private method?

Comment (jbee) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/RedisCache.java:154]: as we use pattern `<region>:<key>` and get `keys( cacheRegion + ":*" )` from redis we know that the keys here must have a `:` and we want just the key part right to it as that is the key used when putting something in the cache.

**Review by maikelarabori — APPROVED** [review]

**Comment by jbee** [comment]
Please do not merge - there is an issue that needs solving.

**Comment by jbee** [comment]
This has been tested by @benguaraldi and it works - merging it.

**Pull request merged**